code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -0,0 +1,558 @@
+	animate(ghost.get_filter("wraith_phase_shift_windup_2"), x = 60*rand() - 30, y = 60*rand() - 30, size=rand()*2.5+0.5, offset=rand(), time = 0.25 SECONDS, loop = -1, flags=ANIMATION_PARALLEL)
+
+	if(!do_after(ghost, WRAITH_PHASE_SHIFT_WINDUP, TRUE, ghost, BUSY_ICON_FRIENDLY)) //Channel time/wind up
+		ghost.visible_message("<span class='xenowarning'>[ghost]'s form abruptly consolidates, returning to normalcy.</span>", \
+		"<span class='xenodanger'>We abort our desynchronization.</span>")
+		ghost.remove_filter("wraith_phase_shift_windup_1")
+		ghost.remove_filter("wraith_phase_shift_windup_2")
+		return fail_activate()
+
+	ghost.remove_filter("wraith_phase_shift_windup_1")
+	ghost.remove_filter("wraith_phase_shift_windup_2")
+
+	playsound(owner, "sound/effects/ghost.ogg", 25, 0, 1)
+
+	addtimer(CALLBACK(src, .proc/phase_shift_warning), WRAITH_PHASE_SHIFT_DURATION * WRAITH_PHASE_SHIFT_DURATION_WARNING) //Warn them when Phase Shift is about to end
+	addtimer(CALLBACK(src, .proc/phase_shift_deactivate), WRAITH_PHASE_SHIFT_DURATION)
+	ghost.add_filter("wraith_phase_shift", 4, list("type" = "blur", 5)) //Cool filter appear
+
+	ghost_movable.generic_canpass = FALSE //So incorporeality is actually checked for collision
+	ghost.status_flags = GODMODE | INCORPOREAL //Become temporarily invulnerable and incorporeal
+	ghost.resistance_flags = RESIST_ALL
+	ghost.density = FALSE
+	ghost.throwpass = TRUE
+	ghost.alpha = WRAITH_PHASE_SHIFT_ALPHA //Become translucent
+
+	starting_turf = get_turf(ghost) //Get our starting turf so we can calculate the stun duration later.
+
+	GLOB.round_statistics.wraith_phase_shifts++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "wraith_phase_shifts") //Statistics
+
+	phase_shift_active = TRUE //Flag phase shift as being active
+	succeed_activate()
+	add_cooldown()
+
+///Warns the user when Phase Shift is about to end.
+/datum/action/xeno_action/phase_shift/proc/phase_shift_warning()
+
+	if(!phase_shift_active) //If phase shift isn't active, cancel out
+		return
+
+	owner.alpha = WRAITH_PHASE_SHIFT_ALPHA * 1.5 //Become less translucent
+
+	to_chat(owner,"<span class='highdanger'>We begin to move back into phase with reality... We can only remain out of phase for [WRAITH_PHASE_SHIFT_DURATION * (1-WRAITH_PHASE_SHIFT_DURATION_WARNING) * 0.1] more seconds!</span>")
+	owner.playsound_local(owner, 'sound/voice/hiss4.ogg', 50, 0, 1)
+
+///Deactivates and turns off the Phase Shift ability/effects
+/datum/action/xeno_action/phase_shift/proc/phase_shift_deactivate()
+	if(!phase_shift_active) //If phase shift isn't active, don't deactivate again; generally here for the timed proc.
+		return
+
+	phase_shift_active = FALSE //Flag phase shift as being off
+
+	var/mob/living/carbon/xenomorph/wraith/ghost = owner
+	var/atom/movable/ghost_movable = owner
```

Comment:
same as before, /xenomorph/wraith/ghost is already an atom/movable

Reply:
this

Reply:
still standing